### ISSUE TITLE
Connect-NsxServer -vCenterServer Failing - 403 Forbidden

### ISSUE THREAD

**Issue by dtrac** [vmware/powernsx#519]
Hi Guys,
New to PowerNSX, but have some experience with PowerShell/PowerCLI.

Downloaded the latest PowerNSX module, and attempting to connect to my vCenter (6.5) & NSX (6.3.4) instance using the administrator@vsphere.local account (which i've granted Enterprise Administrator role in NSX Manager), and seeing the following error:

----------------------------------------

PS C:\Windows\system32> Connect-NsxServer -vCenterServer $vc -Credential $cred
Connection to NSX server <IP Removed!> failed : Invoke-NsxRestMethod : The NSX API response received indicates a failure. 403 : Forbidden : Response Body: 
At C:\Program Files\WindowsPowerShell\Modules\powernsx\PowerNSX.psm1:4844 char:13
+             Throw "Connection to NSX server $NsxServer failed : $_"
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Connection to N...Response Body: :String) [], RuntimeException
    + FullyQualifiedErrorId : Connection to NSX server <IP Removed!> failed : Invoke-NsxRestMethod : The NSX API response received indicates a failure. 403 : Forbidden : Response Body:

-----------------------------------------

I have tested that I can authenticate against the NSX api using Postman with Basic Auth using the same credentials.

Any ideas?

Thanks,

Dan

**Comment by nmbradford** [comment]
That's an odd result given it's a pretty common scenario.  I just sanity tested connection via VC using a stored credential and it works as expected for me.  First - lets validate the obvious :
1. $vc contains an IP or FQDN (not a PowerCLI connection or some other non string object?)
2. $cred contains valid credentials (I have to ask and make sure you didn't get your password wrong)! :) )

Normally, you dont configure specific role access for the sso admin as its usually the account used to register NSX to VC.  Is that not the case here? (Also - role membership is granted in the NSX plugin within vCenter Web UI - is this where you made the change? (you said on the manager itself?)

Can you enable debug ($debugpreference = "continue") and retry the connection.  You will see the URI being hit and failing (I would imagine its https://<manager>:443/api/2.0/nwfabric/features) - can you do the same credential test in postman for that specific URI please and confirm it works?  Also validate that PowerNSX is determining the NSXManager correctly by confirming the ip/name being used in this URI.

Let us know what you find.

**Comment by dtrac** [comment]
Thanks for the reply - I tested using another system and was able to connect, so it looks to be a proxy/firewall issue with the specific host I was using.  I didn't think it was the case initially as I was able to successfully connect with  Test-NetConnection <nsx manager IP> -Port 443 , but trying an Invoke-WebRequest to the specific URI you provided above is failing with our proxy error message.

Thanks for your help.
